pkg/sbproxy/notifications/producer_test.go

```diff
@@ -219,6 +269,69 @@ var _ = Describe("Notifications", func() {
 			})
 		})
 
+		Context("When a server that accept mTLS exists", func() {
+			var tlsServer *httptest.Server
+			var tlsSocketServer *wsServer
+			BeforeEach(func() {
+				tlsServer, tlsSocketServer = newServerWithTLS()
+				producerSettings = &notifications.ProducerSettings{
+					MinPingPeriod:        100 * time.Millisecond,
+					ReconnectDelay:       100 * time.Millisecond,
+					PongTimeout:          20 * time.Millisecond,
+					ResyncPeriod:         300 * time.Millisecond,
+					PingPeriodPercentage: 60,
+					MessagesQueueSize:    10,
+				}
+			})
+
+			AfterEach(func() {
+				tlsServer.Close()
+			})
+
+			When("valid mTLS certificate settings are used", func() {
+
+				BeforeEach(func() {
+					smSettings = &sm.Settings{
+						URL:                  tlsServer.URL,
+						User:                 "admin",
+						Password:             "admin",
+						RequestTimeout:       11 * time.Second,
+						TLSClientCertificate: ClientCertificate,
+						TLSClientKey:         ClientKey,
+						SkipSSLValidation:    true,
+						NotificationsAPIPath: "/v1/notifications",
+					}
+				})
+
+				It("successfully pings the mTLS server using valid client certificate", func(done Done) {
```

Comment:
I don't understand this test, can you explain or maybe add some comments?

Reply:
simplified